### PR TITLE
Integrate Dshuttle into PAI

### PR DESCRIPTION
**Pull request by Binyang2014**
Integral Dshuttle into PAI.
Components:
- dshuttle-master
- dshuttle-worker
- dshuttle-csi

### CODE REVIEW

File: src/dshuttle-master/deploy/delete.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pushd $(dirname "$0") > /dev/null
+
+echo "Call stop to stop all dshuttle-master pod first"
+/bin/bash stop.sh || exit $?
+
+echo "Create dshuttle-master-delete configmap for deleting data on the host"
+kubectl create configmap dshuttle-master-delete --from-file=dshuttle-master-delete/ --dry-run -o yaml | kubectl apply --overwrite=true -f - || exit $?
+
+echo "Create cleaner daemon"
+kubectl apply --overwrite=true -f delete.yaml || exit $?
```
Comment (hzy46) [inline src/dshuttle-master/deploy/delete.sh:29]: Please check whether this command will affect other machines.

File: src/dshuttle-master/deploy/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,185 @@
+#
```
Comment (hzy46) [inline src/dshuttle-master/deploy/log4j.properties:1]: Please make this file a template.

File: src/dshuttle-master/deploy/service.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+cluster-type:
+  - k8s
+
+prerequisite:
+  - cluster-configuration
+  - dshuttle
```
Comment (hzy46) [inline src/dshuttle-master/deploy/service.yaml:23]: Please delete this line.

File: src/dshuttle-master/deploy/delete.sh (hunk 1 above)
Comment (Binyang2014) [reply src/dshuttle-master/deploy/delete.sh:29]: Add `nodeAffinity` in spec. deploy-rules doesn't effect delete script. Seems is by design

File: src/dshuttle-master/deploy/log4j.properties (hunk 2 above)
Comment (Binyang2014) [reply src/dshuttle-master/deploy/log4j.properties]: fixed

File: src/dshuttle-master/deploy/service.yaml (hunk 3 above)
Comment (Binyang2014) [reply src/dshuttle-master/deploy/service.yaml]: fixed

**Review by hzy46 — APPROVED** [review]

**Pull request merged**